piptools/utils.py

```diff
@@ -61,13 +62,30 @@ def make_install_requirement(name, version, extras, constraint=False):
     return InstallRequirement.from_line('{}{}=={}'.format(name, extras_string, str(version)), constraint=constraint)
 
 
+def is_subdirectory(base, directory):
+    """
+    Return True if directory is a child directory of base
+    """
+    base = os.path.join(os.path.realpath(base), '')
+    directory = os.path.join(os.path.realpath(directory), '')
+
+    return os.path.commonprefix([base, directory]) == base
```

Comment:
From os.path docs: «Note that this may return invalid paths because it works a character at a time.»

Python 3.5 added os.path.commonpath to avoid that, which pip-tools may want to backport.